### DISCUSSION TITLE
Graceful shutdown hooks

### DISCUSSION THREAD

**Discussion by belgattitude** [vercel/next.js#20172] [Help]
I'm evaluating the possibility to get more from api-routes or ssr.

I'd like to know if there is a way to make a graceful shutdown and close connections... in case of restart, SIGINT, SIGTERM...

```js
process.on('SIGINT', function() {
   // imaginary
   redis.close();
   pgsql.close()
   prisma.close();
});
```

Is there something we could add to the next.config.js in prod ? Or will it compete with nextjs ones ?

**Comment by stigkj** [comment]
+1

**Comment by jbrardport** [comment]
Set `NEXT_MANUAL_SIG_HANDLE=true` and you should be able to do this
